core/app/models/spree/order_updater.rb

```diff
@@ -183,6 +183,7 @@ def update_item_promotions
         Spree::Config.promotion_chooser_class.new(promotion_adjustments).update
 
         item.promo_total = promotion_adjustments.select(&:eligible?).sum(&:amount)
+        item.save
```

Comment:
Additionally, I think this is an appropriate time to talk about preferring `save!` vs `save`. If we cannot persist an object when we intend to, I cannot see why wouldn't _always_ want an exception. This localized the problem making debugging much easier. I would suggest that all `save`/`update`/`create` type active record persistence methods be replaced with their exception-raising equivalents.